src/service_daemon.rs

```diff
@@ -1920,6 +2148,23 @@ pub enum ServiceEvent {
     SearchStopped(String),
 }
 
+/// All possible events sent to the client from the daemon
+/// regarding host resolution.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum HostnameResolutionEvent {
+    /// Started searching for the ip address of a hostname.
+    SearchStarted(String),
+    /// One or more addresses for a hostname has been found.
+    HostnameAddressesFound(String, HashSet<IpAddr>),
+    /// One or more addresses for a hostname has been removed.
+    HostnameAddressesRemoved(String, HashSet<IpAddr>),
```

Comment:
I felt that `Hostname` prefix could be removed from `HostnameAddressesFound` and `HostnameAddressesRemoved` as the enum name already included `Hostname`. 

That said, I am not super good at naming. So it's only a personal opinion.